doc/4-instrument/fm-esfm.md

```diff
@@ -0,0 +1,119 @@
+- **Left Output (L)** and **Right Output (R)**: toggles output on left/right channels from the operator to the audio output.
+- **Waveform Select (WS)**: changes the waveform of the operator (0 to 7).
+- **Vibrato (VIB)**: makes the operator affected by LFO vibrato.
+- **Vibrato Depth flag (DVB/FMD)**: doubles the effect of LFO vibrato in pitch when set.
+
+### routing controls
+
+- **Output Level (OL)**: sets the output level from this operator to the audio output (0 to 7).
+  - 7 is the loudest level and 1 is the softest, while 0 disables audio output.
+  - this output scaling factor is applied after TL and envelope scaling have been performed.
+  - a change of one unit is about 6 dB.
+- **Modulation Input Level (MI)**: sets the modulation level from the previous operator to this operator (0 to 7).
+  - 7 is the strongest level and 1 is the weakest, while 0 disables modulation.
+  - for operator 1, since there's no previous operator, this controls the **feedback level**.
+  - this modulation scaling factor is applied after the previous operator's TL and envelope scaling have been performed, but is unaffected by OL above.
+  - a change of one unit is about 6 dB.
+
+### table of OL and MI values for obtaining common algorithms
+
+note: MI1 is not included as it is the feedback level.
+
+| algorithm                      | OL1 | MI2 | OL2 | MI3 | OL3 | MI4 | OL4 |
+|--------------------------------|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
+| OPN algorithm **0**            |  0  |  7  |  0  |  7  |  0  |  7  |  7  |
+| OPN algorithm **4**            |  0  |  7  |  7  |  0  |  0  |  7  |  7  |
+| OPN algorithm **6**            |  0  |  7  |  7  |  0  |  7  |  0  |  7  |
+| OPN algorithm **7**            |  7  |  0  |  7  |  0  |  7  |  0  |  7  |
+| OPL3 algorithm **1**           |  7  |  0  |  0  |  7  |  0  |  7  |  7  |
+| OPL3 algorithm **3**           |  7  |  0  |  0  |  7  |  7  |  0  |  7  |
+| OPL3 algorithm **1** (variant) |  0  |  7  |  0  |  7  |  7  |  0  |  7  |
+
+### fixed frequency mode
+
+each operator has a Fixed Frequency mode. once enabled, the operator runs at the specified frequency regardless of the note.
+
+## macros
+
+these macros allow you to control several parameters of FM per tick.
+
+## OP1-OP4 Macros
+
+most parameters are listed above.
+
+### envelope delay macro tricks
+
+due to a quirk in how the envelope delay feature works, the **Envelope Delay** macro can control the operator's key-on status for a limited amount of time after a note is played. a value of 0 represents key-on, while a value of 7 represents key-off.
+
+note that the macro cannot exceed 659.1 ms in length - anything beyond that will be treated as a value of 0.
+
+### operator arpeggio and pitch
+
+among the available macros are **Op. arpeggio** and **Op. pitch**. these work like the **Arpeggio** and **Pitch** macros featured below, but are applied individually over the operator, overriding the **Arpeggio**/**Pitch** macros respectively.
```

Comment:
```suggestion
among the available macros are **Op. arpeggio** and **Op. pitch**. these work like the **Arpeggio** and **Pitch** macros featured below, but are applied to the individual operator, overriding the **Arpeggio**/**Pitch** macros respectively.
```

Reply:
I would also add that despite them overriding main instrument pitch/arp macros, they still obey tune/detune settings set in FM tab